wikipedia/track.py

```diff
@@ -237,11 +213,112 @@ def params(self):
+
+        try:
+            return {
+                "query": f'FROM {self._index_name} METADATA _score | WHERE { query_body } | KEEP title, _score | SORT _score DESC | LIMIT { self._size }',
+            }
+
+        except StopIteration:
+            self._queries_iterator = iter(self._sample_queries)
+            return self.params()
+
+class QueryParamSource(QueryIteratorParamSource):
+    def __init__(self, track, params, **kwargs):
+        super().__init__(track, params, **kwargs)
+        self._index_name = params.get("index", track.indices[0].name if len(track.indices) == 1 else "_all")
+        self._cache = params.get("cache", True)
+        self._query_type = self._params["query-type"]
+
+    def params(self):
+
+        try:
+            query = next(self._queries_iterator)
+            if self._query_type == "query-string":
+                query_body = {"query_string": {"query": query, "default_field": self._params["search-fields"]}}
+            elif self._query_type == "kql":
+                query_body = { "kql": {"query": f'{ self._params["search-fields"] }:"{ query }"'} }
+            elif self._query_type == "match":
+                query_body = {
+                    "bool": {
+                        "should": [
+                            {
+                                "match": {
+                                    "title": query
+                                }
+                            },
+                            {
+                                "match": {
+                                    "content": query
+                                }
+                            }
+                        ]
+                    }
+                }
+            elif self._query_type == "term":
+                query_body = {
+                    "bool": {
+                        "should": [
+                            {
+                                "term": {
+                                    "title": query
+                                }
+                            },
+                            {
+                                "term": {
+                                    "content": query
+                                }
+                            }
+                        ]
+                    }
+                }
+            else:
+                raise ValueError("Unknown query type: " + self._query_type)
+
+        except StopIteration:
+            self._queries_iterator = iter(self._sample_queries)
+            return self.params()
+
+        return {
+            "body": {
+                "_source": {"includes": ["title"]},
+                "query": query_body,
+            },
+            "size": self._params["size"],
+            "index": self._index_name,
+            "cache": self._cache,
+        }
+
+
 def register(registry):
-    registry.register_param_source("query-string-search", QueryParamSource)
```

Comment:
Is there any impact from this renaming?

Reply:
The `param-source` attribute for the `query-string-search` operation has been changed - no other changes are needed as just the param source is changed and not the actual query name.